apps/accounts/views.py

```diff
@@ -49,3 +65,129 @@ def get_auth_token(request):
+        except InviteUserToChallenge.DoesNotExist:
+            user, created = User.objects.get_or_create(
+                username=email, email=email
+            )
+            if created:
+                EmailAddress.objects.create(
+                    user=user, email=email, primary=True, verified=True
+                )
+            invitation_key = uuid.uuid4()
+            invitation_status = "pending"
+            data = {
+                "email": email,
+                "invitation_key": str(invitation_key),
+                "status": invitation_status,
+                "challenge": challenge.pk,
+                "user": user.pk,
+                "invited_by": challenge_host.pk,
+            }
+            serializer = InviteUserToChallengeSerializer(data=data)
+
+            if serializer.is_valid():
+                serializer.save()
+            else:
+                invalid_emails.append(email)
+
+        sender_email = settings.CLOUDCV_TEAM_EMAIL
+        url = "{}/accept/invitation/{}/".format(
+            settings.EVALAI_HOST_URL, invitation_key
+        )
+        context = {"title": challenge.title, "url": url}
+
+        if email not in invalid_emails:
+            send_email(
+                subject,
+                "challenge_invitation_email.html",
+                sender_email,
+                email,
+                context,
+            )
+
+    if len(users_email) == len(invalid_emails):
+        response_data = {"error": "Please enter correct email ids"}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    response_data = {
+        "success": "The invitation to register for {} is sent".format(
+            challenge.title
+        ),
+        "invalid_emails": invalid_emails,
+    }
+    return Response(response_data, status=status.HTTP_200_OK)
+
+
+@api_view(["GET", "PUT"])
+@throttle_classes([UserRateThrottle])
+@permission_classes(())
+def accept_challenge_invitation(request, invitation_key):
+    try:
+        invitation = InviteUserToChallenge.objects.get(
+            invitation_key=invitation_key
+        )
+    except InviteUserToChallenge.DoesNotExist:
+        response_data = {
+            "error": "The invitation with key {} doesn't exist".format(
+                invitation_key
+            )
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if request.method == "GET":
+        serializer = InviteUserToChallengeSerializer(invitation)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    elif request.method == "PUT":
+        serializer = AcceptChallengeInvitationSerializer(
+            invitation.user, data=request.data
+        )
+        if serializer.is_valid():
+            serializer.save()
+            invitation.user.set_password(serializer.data.get("password"))
+            invitation.user.save()
+            invitation.status = "accepted"
```

Comment:
Can we change this to `invitation.status = UserInvitation.ACCEPTED`

Reply:
Similar hardcoding is done in some previous lines. Please change it to variable as I have mentioned in the previous comment.